website/addons/owncloud/static/owncloudNodeConfig.js

```diff
@@ -0,0 +1,315 @@
+            return self.nodeHasAuth() && !self.validCredentials();
+        });
+        self.showCredentials = ko.pureComputed(function() {
+            return (self.credentialsChanged() && self.userIsOwner()) ||
+                (!self.userHasAuth() && !self.nodeHasAuth() );
+        });
+        self.showCredentialInput = ko.pureComputed(function(){
+            return Boolean(self.selectedHost());
+        });
+
+        // Overrides
+        var defaults = {
+            onPickFolder: function(evt, item) {
+                evt.preventDefault();
+                var name = item.data.path !== '/' ? item.data.path : '/ (Full ' + self.addonName + ')';
+                self.selected({
+                    name: name,
+                    path: item.data.path,
+                    id: item.data.id
+                });
+                self.folderName(name);
+                return false; // Prevent event propagation
+            },
+            connectAccount: function() {
+                window.location.href = this.urls().auth;
+            },
+            decodeFolder: function(folder_name) {
+                return folder_name;
+            }
+        };
+        // Overrides
+        self.options = $.extend({}, defaults, opts);
+        // Treebeard config
+        self.treebeardOptions = $.extend(
+            {},
+            OauthAddonFolderPicker.prototype.treebeardOptions,
+            {
+                onPickFolder: function(evt, item) {
+                    return this.options.onPickFolder.call(this, evt, item);
+                }.bind(this),
+                resolveLazyloadUrl: function(item) {
+                    return item.data.urls.folders;
+                },
+                decodeFolder: function(item) {
+                    return this.options.decodeFolder.call(this, item);
+                }.bind(this)
+
+            }
+        );
+
+        self.credentialsChanged = ko.pureComputed(function() {
+            return self.nodeHasAuth() && !self.validCredentials();
+        });
+        self.showCredentialInput = ko.pureComputed(function() {
+            return (self.credentialsChanged() && self.userIsOwner()) ||
+                (!self.userHasAuth() && !self.nodeHasAuth() && self.loadedSettings());
+        });
+
+        self.showSettings = ko.pureComputed(function() {
+            return self.nodeHasAuth() && self.validCredentials();
+        });
+        self.showImport = ko.pureComputed(function() {
+            return self.userHasAuth() && !self.nodeHasAuth() && self.loadedSettings();
+        });
+        self.showCredentialCreateButton = ko.pureComputed(function() {
+            return !self.userHasAuth() && !self.nodeHasAuth() && self.loadedSettings();
+        });
+
+        // Flashed messages
+        self.message = ko.observable('');
+        self.messageClass = ko.observable('text-info');
+
+        // Update above observables with data from the server
+        $.ajax({
```

Comment:
You should be overriding `updateFromData` rather than doing this here. All that actually needs to be done is overriding `applySettings`, but it seems to be less extensible than it should be, as that function is defined within `updateFromData`.